### PR TITLE
feat: `mongodbatlas_search_deployment` data source with acceptance tests and docs

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Jira ticket: [INTMDB-1273](https://jira.mongodb.org/browse/INTMDB-1273)

Currently open as draft pull request as we have to make use of a dev preview of the Atlas SDK.

This PR includes:

- Implementation of `mongodbatlas_search_deployment` data source
- acceptance tests
- documentation

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/fw_data_source_mongodbatlas_search_deployment.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	ProjectID   types.String `tfsdk:"project_id"`
+	Specs       types.List   `tfsdk:"specs"`
+	StateName   types.String `tfsdk:"state_name"`
+}
+
+type SearchDeploymentDS struct {
+	DSCommon
+}
+
+func (d *SearchDeploymentDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
```
Comment (marcosuma) [inline mongodbatlas/fw_data_source_mongodbatlas_search_deployment.go:34]: this is more of a generic question: in the context of a data source I don't understand the concept of `Computed`: aren't they all computed?

On the other side I'd imagine `Required` means that if we don't fill it up with the backend response terraform would raise an error, but why do we really need to set it as required? Couldn't we just leave it as Optional?

File: mongodbatlas/fw_resource_mongodbatlas_search_deployment.go [hunk 2]
```diff
@@ -31,12 +31,13 @@ var _ resource.ResourceWithImportState = &SearchDeploymentRS{}
 
 const (
 	searchDeploymentDoesNotExistsError = "ATLAS_FTS_DEPLOYMENT_DOES_NOT_EXIST"
+	searchDeploymentName               = "search_deployment"
```
Comment (marcosuma) [inline mongodbatlas/fw_resource_mongodbatlas_search_deployment.go:34]: nit: this looks like a separate PR? no need to do it now, but consider it for the next time

File: mongodbatlas/fw_data_source_mongodbatlas_search_deployment.go (hunk 1 above)
Comment (AgustinBettati) [reply mongodbatlas/fw_data_source_mongodbatlas_search_deployment.go:34]: > in the context of a data source, aren't they all computed?

Not exactly. When using a data source there are some attributes that need to be provided by the user in order to identify the resource that will be fetched. In this case the resource is identified through the `project_id` and `cluster_name`, which is why they are marked as required.
You will see this is also clarified in the documentation of the data source, separating the concept of arguments (attributes that have to be defined), and then an attributes sections for values that are exported (or as we call computed).

File: mongodbatlas/fw_resource_mongodbatlas_search_deployment.go (hunk 2 above)
Comment (AgustinBettati) [reply mongodbatlas/fw_resource_mongodbatlas_search_deployment.go:34]: Yes true. Tried to keep this PR as clean as possible with respect to the resource implementation, but you usually some pending details always get adjusted when implementing the data source.

**Comment by AgustinBettati** [comment]
merged in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1633

**Pull request closed without merging**